docs/HowToGuides/RunningIncludeWhatYouUse.md

```diff
@@ -0,0 +1,289 @@
+   ```
+2. Build IWYU
+   ```
+   cmake --build .
+   ```
+3. Create the `lib` directory if missing.
+   ```
+   file lib || ln -s ../../build/Ninja-ReleaseAssert/llvm-macosx-x86_64/lib lib
+   ```
+   This is needed for IWYU to find necessary Clang headers.
+4. Spot check IWYU for a basic C example.
+   ```
+   echo '#include <stdint.h>' > tmp.c
+   ./bin/include-what-you-use tmp.c -E -o /dev/null \
+     -I "$(xcrun --show-sdk-path)/usr/include"
+   rm tmp.c
+   ```
+   You should see output like:
+   ```
+   tmp.c should add these lines:
+
+   tmp.c should remove these lines:
+   - #include <stdint.h>  // lines 1-1
+
+   The full include-list for tmp.c:
+   ---
+   ```
+5. Spot check IWYU for a basic C++ example. Notice the extra C++-specific
+   include path.
+   ```
+   echo '#include <string>\n#include <cmath>' > tmp.cpp
+   ./bin/include-what-you-use tmp.cpp -E -o /dev/null \
+     -I "$(clang++ -print-resource-dir)/../../../include/c++/v1" \
+     -I "$(xcrun --show-sdk-path)/usr/include"
+   rm tmp.cpp
+   ```
+   You should see output like:
+   ```
+   tmp.cpp should add these lines:
+
+   tmp.cpp should remove these lines:
+   - #include <cmath>  // lines 2-2
+   - #include <string>  // lines 1-1
+
+   The full include-list for tmp.cpp:
+   ---
+   ```
+6. Leave the build directory.
+   ```
+   cd ..
+   ```
+
+## Running IWYU
+
+1. Create a directory, say `iwyu/scripts`, and copy the following script there.
+
+   ```
+   #!/usr/bin/env bash
+
+   # iwyu_run.sh
+   set -eu
+
+   SWIFT_PROJECT_DIR="$HOME/swift-project"
+   SWIFT_BUILD_DIR="$SWIFT_PROJECT_DIR/build/Ninja-ReleaseAssert/swift-macosx-x86_64"
+
+   pushd "$SWIFT_BUILD_DIR"
+
+   if [ -f original_compile_commands.json ]; then
+       mv original_compile_commands.json compile_commands.json
+   fi
+
+   # HACK: The additional include path needs to be added before other include
+   # paths, it doesn't seem to work if we add it at the end.
+   # It is ok to rely on the presence of `-D__STDC_LIMIT_MACROS` since
+   # all compiler invocations have that flag, and experimentally, that
+   # flag appears before include paths.
+   # NOTE: IWYU doesn't work for ObjC, so filter out ObjC files.
+   ( EXTRA_CXX_INCLUDE_DIR="$(clang++ -print-resource-dir)/../../../include/c++/v1";
+     cat compile_commands.json \
+    | jq '[.[] | select(.file | test(".mm" | ".m") | not) | {directory: .directory, command: (.command + " -Wno-everything -ferror-limit=1"), file: .file}]' \
+    | sed -e "s|-D__STDC_LIMIT_MACROS |-D__STDC_LIMIT_MACROS -I $EXTRA_CXX_INCLUDE_DIR |" \
+   ) > filtered_compile_commands.json
+
+   mv compile_commands.json original_compile_commands.json
+   mv filtered_compile_commands.json compile_commands.json
+
+   mkdir -p "$SWIFT_PROJECT_DIR/iwyu/logs"
+
+   ( PATH="$SWIFT_PROJECT_DIR/iwyu/build/bin:$PATH"; \
+     "$SWIFT_PROJECT_DIR/iwyu/include-what-you-use/iwyu_tool.py" -p "$SWIFT_BUILD_DIR"
+   ) | tee "$SWIFT_PROJECT_DIR/iwyu/logs/suggestions.log"
+
+   popd
+
+   ```
```

Comment:
Does it make sense to use `iwyu_tool.py` from iwyu instead of a custom script?

Reply:
The custom script is filtering out ObjC files and suppressing warnings in the output, in case there are any. I don't see how using `iwyu_tool.py` directly will do the same.

Reply:
I think that IWYU should be able to handle ObjC/ObjC++ as long as you are either on macOS and have the SDK or do not use any of the ObjC headers.  That is, I would have expected it to work on the ObjC/ObjC++ sources we have in Swift.

Reply:
Good point about the failure message, I'll add that so someone can pattern-match in case someone runs into it. IIRC I hit [this error](https://github.com/include-what-you-use/include-what-you-use/blob/b4838853e5c906854a616d9e975b53c164950174/iwyu.cc#L2103). There are also similar notes [elsewhere](https://github.com/include-what-you-use/include-what-you-use/blob/b4838853e5c906854a616d9e975b53c164950174/iwyu_ast_util.cc#L1196).

Reply:
Interesting, but that sounds like a good way to handle this then.